lib/ansible/modules/cloud/azure/azure_rm_manageddisk.py

```diff
@@ -460,6 +460,8 @@ def get_managed_disk(self):
             self.log('Did not find managed disk')
 
     def is_attach_caching_option_different(self, vm_name, disk):
+        if self.attach_caching is None:
```

Comment:
Instead of adding check like this, I think just init `attach_caching` as '' will be better. There should be only three options (`['', 'read_only', 'read_write']`) for it.

Reply:
The module itself supports all 3 choices, but API returns only `read_only` or `read_write`, so empty option is always change in next runs.
However, you're right - if one set attach_caching to empty string, my fix won't work. I'll fix it.

Reply:
I've double check code and empty string means "no caching", so it shouldn't be treated as a no-change. The https://github.com/ansible/ansible/blob/fae58a588afe8b24cf59ea0ddce39927d80666be/lib/ansible/modules/cloud/azure/azure_rm_manageddisk.py#L471 take care of it.
Not setting `attach_caching` (keeping it None, no changes) *is* different than setting it to empty string (disable caching), so I believe current fix is sufficient.